RFS/src/test/java/org/opensearch/migrations/bulkload/version_es_6_8/OpenSearchClient_ES_6_8_Test.java

```diff
@@ -0,0 +1,117 @@
+import reactor.core.publisher.Mono;
+
+import static org.hamcrest.CoreMatchers.containsString;
+import static org.hamcrest.CoreMatchers.not;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.spy;
+import static org.mockito.Mockito.verifyNoInteractions;
+import static org.mockito.Mockito.when;
+import static org.mockito.Mockito.withSettings;
+import static org.opensearch.migrations.bulkload.http.BulkRequestGenerator.itemEntry;
+
+@ExtendWith(MockitoExtension.class)
+class OpenSearchClient_ES_6_8_Test {
+    private static final ObjectMapper OBJECT_MAPPER = JsonMapper.builder()
+        .enable(StreamReadFeature.INCLUDE_SOURCE_IN_LOCATION)
+        .build();
+
+    @Mock(strictness = Strictness.LENIENT)
+    RestClient restClient;
+
+    @Mock
+    ConnectionContext connectionContext;
+
+    @Mock
+    FailedRequestsLogger failedRequestLogger;
+
+    OpenSearchClient openSearchClient;
+
+    @BeforeEach
+    void beforeTest() {
+        openSearchClient = spy(new OpenSearchClient_ES_6_8(restClient, failedRequestLogger, Version.fromString("ES_6_8")));
+    }
+
+    @SneakyThrows
+    @Test
+    void testCreateIndexWithIncludeTypeNameFalse() {
+        // Setup
+        var checkIfExistsResponse = new HttpResponse(404, "", null, "does not exist");
+        var createdItemRawJson = "{\"created\":\"yup!\"}";
+        var createItemResponse = new HttpResponse(200, "", null, createdItemRawJson);
+
+        when(restClient.getAsync(any(), any())).thenReturn(Mono.just(checkIfExistsResponse));
+        when(restClient.putAsync(any(), any(), any())).thenReturn(Mono.just(createItemResponse));
+
+        // Action
+        var rawJson = "{ }";
+        var body = (ObjectNode) OBJECT_MAPPER.readTree(rawJson);
+        var result = openSearchClient.createIndex("indexName", body, mock(ICheckedIdempotentPutRequestContext.class));
+
+        // Assertions
+        assertThat(result.get().toPrettyString(), containsString(rawJson));
+        // The interface is to send back the passed json if on success
+        assertThat(result.get().toPrettyString(), not(containsString(createdItemRawJson)));
+
+        Mockito.verify(restClient).putAsync("indexName?include_type_name=false", "{}", null);
+    }
+
+    @Test
+    void testBulkRequestHasType_doc() {
+        // This test gets stuck! I can't figure out why. Somewhere in the `sendBulkRequest`, it's presumably waiting for
+        // an async to return, but I can't figure out what. Running in debug hasn't turned anything up.
+        var docId1 = "tt1979320";
+        var docId2 = "tt0816711";
+
+        var successResponse = bulkItemResponse(false, List.of(itemEntry(docId1), itemEntry(docId2)));
+        var bulkDocs = List.of(createBulkDoc(docId1), createBulkDoc(docId2));
+        when(restClient.postAsync(any(), any(), any())).thenReturn(Mono.just(successResponse));
```

Comment:
This was causing the test to hang be cause postAsync(3 args) was never called, whereas postAsync(4 args) is on the sendBulkRequest method.  
```suggestion
        when(restClient.postAsync(any(), any(), any(), any())).thenReturn(Mono.just(successResponse));
```